internal/buffer/buffer.go

```diff
@@ -710,57 +710,59 @@ func (b *Buffer) UpdateRules() {
 	var header *highlight.Header
 	// search for the syntax file in the user's custom syntax files
 	for _, f := range config.ListRealRuntimeFiles(config.RTSyntax) {
-		data, err := f.Data()
-		if err != nil {
-			screen.TermMessage("Error loading syntax file " + f.Name() + ": " + err.Error())
-			continue
-		}
-
-		header, err = highlight.MakeHeaderYaml(data)
-		if err != nil {
-			screen.TermMessage("Error parsing header for syntax file " + f.Name() + ": " + err.Error())
-			continue
-		}
-
-		matchedFileType := false
-		matchedFileName := false
-		matchedFileHeader := false
-
-		if ft == "unknown" || ft == "" {
-			if header.MatchFileName(b.Path) {
-				matchedFileName = true
-			}
-			if len(fnameMatches) == 0 && header.MatchFileHeader(b.lines[0].data) {
-				matchedFileHeader = true
-			}
-		} else if header.FileType == ft {
-			matchedFileType = true
-		}
-
-		if matchedFileType || matchedFileName || matchedFileHeader {
-			file, err := highlight.ParseFile(data)
+		if f.Name() != "default" {
```

Comment:
The idea behind it was, that I ignore the possible overriden default in the first stage and only consider it, when it's really necessary.
Anyway, your hint is right in the moment the change shall be review-friendly.

Reply:
> The idea behind it was, that I ignore the possible overriden default in the first stage and only consider it, when it's really necessary.

Yeah I get that, but what does it have to do with the question "why not use `continue`"? :)

So, why not use `continue`? It not just makes the change more review-friendly but also the code a bit more readable (no extra level of nesting, no large `if` block etc).

(Maybe I've misunderstood you and you're already going to change it to use `continue`?)

Reply:
Woah, I've been blind while reading...

```
if f.Name() == "default" {
	continue
}
```

...since you really gave that hint before. Damn!
I will definitely apply. Don't ask me where it's coming from, maybe it's the style we've to apply at work...